pkgs/servers/matrix-synapse/matrix-appservice-irc/default.nix

```diff
@@ -24,6 +24,24 @@ buildNpmPackage rec {
     python3
   ];
 
+  # Clean up test artifacts.
+  postCheck = ''
+    rm -r spec-db
+  '';
+
+  # We need the crypto SDK to be in place for tests...
+  preCheck = ''
+    rm -r node_modules/@matrix-org/matrix-sdk-crypto-nodejs
+    cp -r --no-preserve=mode ${matrix-sdk-crypto-nodejs}/lib/node_modules/@matrix-org/matrix-sdk-crypto-nodejs node_modules/@matrix-org/
+  '';
+
+  # ...but `npm pack` also checks its integrity/version...
+  preInstall = ''
+    touch node_modules/.package-lock.json
+    sed -i "s/0.1.0-beta.2/0.1.0-beta.1/g" node_modules/@matrix-org/matrix-sdk-crypto-nodejs/package.json
+  '';
```

Comment:
Could we just shuffle the original directory into a hidden directory in `preCheck` and link in the crypto SDK, and then restore the original (now hidden) directory in `postCheck`?

Magic versions in a `sed` regex (unescaped) during `preInstall` seems like it would be brittle, so it would be nice to avoid it if you haven't tried something like the above yet